From 346e85c1ae98d82e209053012a24cd477b2a8c06 Mon Sep 17 00:00:00 2001
From: Adriana Reus <adriana.reus@intel.com>
Date: Fri, 30 Aug 2013 14:04:09 +0300
Subject: SQUISH Check file object for null before closing

Please sqush with the following patch on the next rebase:
6f127c0 Signing ability in mkbootimg

Change-Id: I64aa7300b09a6b0e634fd0cd298cda030833d9e0
Signed-off-by: Adriana Reus <adriana.reus@intel.com>
---
 mkbootimg/mkbootimg.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/mkbootimg/mkbootimg.c b/mkbootimg/mkbootimg.c
index e49c1b9..2a974be 100644
--- a/mkbootimg/mkbootimg.c
+++ b/mkbootimg/mkbootimg.c
@@ -218,7 +218,8 @@ static int read_write_data(void *outdata, size_t outsz,
         }
 
 rw_done:
-        fclose(fp);
+        if (fp)
+            fclose(fp);
         return result;
     }
 }
-- 
1.7.4.1

